### ISSUE TITLE
Conditionally disable compositing

### ISSUE THREAD

**Issue by SubhadeepJasu** [elementary/gala#1179]
## Problem
- Use Case 1:
The problem happens mainly while gaming. Gala uses a lot of resources especially GPU usage which reduces gaming performance a lot.

- Use Case 2:
When a computer doesn't have proper graphics support or lack of any 3D processing GPU. (For example, my old laptop's 3D GPU was removed due to damage, or machines with just a weak Intel HD GPU)

In both cases, disabling compositing and relying on xrender would boost performance (and in case 2 make the desktop actually usable)

## Possible Solutions
- Disable compositing when an application (by matching process name) goes fullscreen.
- Allow disabling compositing via console commands. So for 3rd party platform games like ones downloaded from Steam can disable compositing while they are running. This is done by using Steams launch commands functionality which for example for XFCE desktop is written as follows `xfconf-query -c xfwm4 -p /general/use_compositing -s false; %command%; xfconf-query -c xfwm4 -p /general/use_compositing -s true`
![Steam Screenshot](https://user-images.githubusercontent.com/20795161/125152930-e1bee100-e16d-11eb-9ff0-636d8428e8fc.png)
- 3rd party elementary apps and games may request gala to turn off compositing via D-Bus.
- Providing a sort of performance mode/power saving mode/gaming mode, which disables compositing.

## Possible Design Issues
System theme may look broken without compositing. I don't know how many essential systems rely on compositing in Gala. There can be fall back systems designed which can re-enable such functionality in lack of compositing.

**Comment by lenemter** [comment]
I don't think we can implement this very easily? The only think we can do is to disable animations but that doesn't use a lot of resources
